### PR TITLE
Implement Polling Navbar

### PR DESCRIPTION
**Pull request by amitark7**

### CODE REVIEW

File: src/component/Navbar.js [hunk 1]
```diff
@@ -1,31 +1,73 @@
...
+          } w-full md:pb-0 md:static flex-col md:flex md:flex-row gap-3 text-[14px] md:text-lg font-semibold cursor-pointer md:items-center`}
+          onClick={() => {
+            setShowMenu(false);
+            setShowLogoutModal(false);
+          }}
+        >
+          <li>
+            <Link to={"/polling"}>Polls</Link>
+          </li>
+          {userData.roleId === 2 && (
```
Comment (yuvrajKaexpertise) [inline src/component/Navbar.js]: instead of direct 2 value use a constant file then use.

File: src/component/Navbar.js [hunk 2]
```diff
@@ -1,31 +1,73 @@
...
+              </li>
+              <li>
+                <Link to={"/users"}>List Users</Link>
+              </li>
+            </>
+          )}
+        </ul>
+        <div
+          className="flex items-center gap-2"
+          onClick={() => setShowLogoutModal(!showLogoutModal)}
```
Comment (yuvrajKaexpertise) [inline src/component/Navbar.js]: Is it for particulat logout modal, I think you are showing name email also so take a state vairable according to it.

File: src/component/Navbar.js [hunk 3]
```diff
@@ -1,31 +1,73 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import { FaUserCircle } from "react-icons/fa";
 import { Link } from "react-router-dom";
+import { FaBars } from "react-icons/fa";
+
+const Navbar = ({ onLogout }) => {
+  const [userData, setUserData] = useState({});
+  const [showLogoutModal, setShowLogoutModal] = useState(false);
+  const [showMenu, setShowMenu] = useState(false);
```
Comment (yuvrajKaexpertise) [inline src/component/Navbar.js]: which menu?

File: src/component/Navbar.js [hunk 4]
```diff
@@ -1,31 +1,69 @@
...
 import { FaUserCircle } from "react-icons/fa";
 import { Link } from "react-router-dom";
+import { FaBars } from "react-icons/fa";
+import { navbarRoute } from "../utils/navbarData";
+
+const Navbar = ({ onLogout }) => {
+  const [userData, setUserData] = useState({});
+  const [showProfile, setShowProfile] = useState(false);
+  const [showNavbarMenu, setShowNavbarMenu] = useState(false);
+  const ADMIN_ID=2
```
Comment (yuvrajKaexpertise) [inline src/component/Navbar.js]: create a seprate file for this and then import it .

File: src/component/Navbar.js [hunk 5]
```diff
@@ -1,31 +1,69 @@
...
-          <Link to={"/users"}>List Users</Link>
-        </li>
-      </ul>
-      <div className="flex items-center gap-2">
-        <div className="text-4xl cursor-pointer">
-          <FaUserCircle />
+    <div className="relative">
+      <div className="flex justify-between items-center px-4 sm:px-8 md:px-10 py-2 bg-black text-white">
+        <div className="md:hidden block text-xl">
+          <FaBars onClick={() => setShowNavbarMenu(!showNavbarMenu)} />
```
Comment (yuvrajKaexpertise) [inline src/component/Navbar.js:33]: user profile menu

File: package.json [hunk 6]
```diff
@@ -14,6 +14,7 @@
     "react-redux": "^9.1.1",
     "react-router-dom": "^6.22.3",
     "react-scripts": "5.0.1",
+    "surge": "^0.23.1",
```
Comment (meanstack9716) [inline package.json]: why do you need surge in package @amitark7 ?

File: src/redux/reducer/rollListFetchReducer.js [hunk 7]
```diff
@@ -4,7 +4,12 @@ import axios from "axios";
 export const getRoleList = createAsyncThunk("role/roleFetch", async () => {
   try {
     const response = await axios.get(
-      `${process.env.REACT_APP_BASE_URL}role/list`
+      `${process.env.REACT_APP_BASE_URL}role/list`,
+      {
+        headers:{
```
Comment (meanstack9716) [inline src/redux/reducer/rollListFetchReducer.js]: created interceptor for headers @amitark7

File: src/utils/constantData.js [hunk 8]
```diff
@@ -0,0 +1 @@
+export const ADMIN_ID=2;
```
Comment (meanstack9716) [inline src/utils/constantData.js]: format code @amitark7

File: src/utils/navbarData.js [hunk 9]
```diff
@@ -0,0 +1,13 @@
+export const navbarRoute=[
+  {
+    name:"Add Poll",
```
Comment (meanstack9716) [inline src/utils/navbarData.js]: format code @amitark7

**Review by meanstack9716 — CHANGES_REQUESTED** [review]

File: src/App.js [hunk 10]
```diff
@@ -13,7 +13,12 @@ import PrivateRoute from "./utils/PrivateRoute";
 function App() {
   const [isLoggedIn, setIsLoggedIn] = useState(false);
   const location = useLocation();
-
+  const navigate = useNavigate();
+  const onLogout = () => {
```
Comment (meanstack9716) [inline src/App.js]: is there a reason this function is not in the navbar code? @amitark7

**Review by meanstack9716 — CHANGES_REQUESTED** [review]

**Pull request merged**